src/main/java/org/broadinstitute/hellbender/utils/downsampling/ReservoirDownsampler.java

```diff
@@ -118,8 +151,9 @@ public boolean hasFinalizedItems() {
         if (hasFinalizedItems()) {
             // pass reservoir by reference rather than make a copy, for speed
             final List<GATKRead> downsampledItems = reservoir;
+            final int depth = totalReadsSeen;
             clearItems();
-            return downsampledItems;
+            return depth < depthToIgnoreLocus ? downsampledItems : Collections.emptyList();
```

Comment:
You're right that we could speed the downsampling with this kind of shortcut, but this is an edge case and almost all the benefit comes from skipping the reads downstream of the downsampler.  The cost of downsampling itself is really small.  Now, if somehow we could downsample in such a way that we could actually make reading the bam file cheaper we would want to short-circuit as much as possible to avoid unnecessary I/O.  Since the reads are already parsed into memory before downsampling there's not a lot to be gained.

Of course, there's not a lot to be lost either, but for me the tie-breaker is keeping changes to code shared with HaplotypeCaller as minimal as possible.